File: C#/DesignPatterns/Facade/Memory.cs
namespace Facade
{
	public class Memory
	{
		private Dictionary<long, byte[]> memory = new Dictionary<long, byte[]>();

		public void Load(long position, byte[] data)
		{
			Console.WriteLine($"Memory starts loading at position {position} with data length of {data.Length}.");
			memory[position] = data;
		}

		public byte[]? GetData(long position)
		{
			if (memory.TryGetValue(position, out var data))
			{
				Console.WriteLine($"Data retrieved from memory at position {position}.");
				return data;
			}

			Console.WriteLine($"No data available in memory at position {position}.");
			return null;
		}

		public void Clear()
		{
			memory.Clear();
		}
	}
}

File: Java/DesignPatterns/src/main/java/Facade/Memory.java
package Facade;

import java.util.HashMap;
import java.util.Map;

class Memory {
    private Map<Long, byte[]> memory = new HashMap<>();

    public void load(long position, byte[] data) {
        System.out.println("Memory starts loading at position " + position + " with data length of " + data.length + ".");
        memory.put(position, data);
    }

    public byte[] getData(long position) {
        byte[] data = memory.get(position);
        if (data != null) {
            System.out.println("Data retrieved from memory at position " + position + ".");
            return data;
        }

        System.out.println("No data available in memory at position " + position + ".");
        return null;
    }

    public void clear() {
        memory.clear();
    }
}
